### ISSUE TITLE
x/tools/go/analysis, syscall: ptrace redeclared in this block

### ISSUE THREAD

**Issue by ldez** [golang/go#44459]
### What version of Go are you using (`go version`)?

<pre>
$ go version
go version go1.16 linux/amd64
</pre>

### Does this issue reproduce with the latest release?

yes

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
$ go env

</pre></details>

### What did you do?

Currently, I'm only able to reproduce inside my IDE (goland).

I'm still trying to reproduce outside of goland.

I'm thinking that the build tag is missing in `syscall/ptrace_ios.go`

I created a minimal reproducible (inside goland) project:

<details>

- [ptrace-bug.zip](https://github.com/golang/go/files/6015745/ptrace-bug.zip)

```
.
├── foo_test.go
├── go.mod
├── go.sum
└── testdata
    └── src
        └── a
            └── a.go
```

foo_test.go:
```go
package sandbox

import (
	"testing"

	"golang.org/x/tools/go/analysis"
	"golang.org/x/tools/go/analysis/analysistest"
)

func TestBug(t *testing.T) {
	ana := &analysis.Analyzer{
		Name:     "ptrace-bug",
		Doc:      "example",
		Run: func(pass *analysis.Pass) (interface{}, error) {
			return nil, nil
		},
		Requires: nil,
	}

	testdata := analysistest.TestData()
	analysistest.Run(t, testdata, ana, "a")
}
```

a.go:

```go
package a

import (
	"time"
)

func validCases() {
	_ = time.Second * 30
}
```

</details>

### What did you expect to see?

```
GOROOT=/home/ldez/.gvm/gos/go1.16 #gosetup
GOPATH=/home/ldez/sources/go #gosetup
/home/ldez/.gvm/gos/go1.16/bin/go test -json ./...
=== RUN   TestBug
--- PASS: TestBug (0.52s)
PASS
ok  	github.com/golangci/sandbox	0.524s

Process finished with exit code 0
```

### What did you see instead?

```
GOROOT=/home/ldez/.gvm/gos/go1.16 #gosetup
GOPATH=/home/ldez/sources/go #gosetup
/home/ldez/.gvm/gos/go1.16/bin/go test -json ./...
=== RUN   TestBug
/home/ldez/.gvm/gos/go1.16/src/syscall/zsyscall_linux_amd64.go:178:6: ptrace redeclared in this block
/home/ldez/.gvm/gos/go1.16/src/syscall/ptrace_ios.go:10:6: 	other declaration of ptrace
    analysistest.go:294: error analyzing ptrace-bug@a: analysis skipped due to errors in package
--- FAIL: TestBug (0.69s)

FAIL
FAIL	github.com/golangci/sandbox	0.694s

Process finished with exit code 1
```

**Comment by ldez** [comment]
I'm able to create a reproducible example without the IDE.

with go1.16:

```console
$ go run .
-----------------
exit status 1
-----------------
/home/ldez/.gvm/gos/go1.16/src/syscall/zsyscall_linux_amd64.go:178:6: ptrace redeclared in this block
/home/ldez/.gvm/gos/go1.16/src/syscall/ptrace_ios.go:10:6:      other declaration of ptrace
--- FAIL: TestBug (0.78s)
    analysistest.go:294: error analyzing ptrace-bug@a: analysis skipped due to errors in package
FAIL
FAIL    github.com/golangci/sandbox     0.784s
FAIL
```

with go1.15:
```console
 go run .
-----------------
<nil>
-----------------
ok      github.com/golangci/sandbox     0.436s
```

[ptrace-bug-repo.zip](https://github.com/golang/go/files/6016145/ptrace-bug-repo.zip)

**Comment by ianlancetaylor** [comment]
@gopherbot Please open backport to 1.16.

The 1.16 release can't be analyzed by tools built with earlier versions of Go, because those earlier releases don't recognize that "ios" is now a valid GOOS value.  The fix is to use an explicit build tag.  We should backport that fix to the 1.16 release branch.
